doc/source/blueprint_cce_vault.rst

```diff
@@ -0,0 +1,592 @@
+=============================================
+(No) Secret Zero with CCE and Hashicorp Vault
+=============================================
+
+With centralized secret management it becomes unnecessary to keep secrets used
+by various application spreaded across DevOps environments. This helps to close
+some security attack vectors, but usually introduces a problem of the so-called
+Secret Zero as a key to the key storage.
+
+This article demonstrates how to utilize Infrastructure authorization of the
+HashiCorp Vault with CCE on an example of deploying Zookeeper cluster with
+enabled TLS protection. Certificates for Zookeeper are stored in Vault where
+they oblige required practices (rotation, audit, etc).
+
+Overview
+========
+
+.. graphviz:: dot/cce_vault_overview.dot
+   :caption: CCE and Vault overview
+   :layout: dot
+
+TLS secrets are kept in the Vault. They are being read by Vault Agent component
+running in the every pod of Zookeeper service, where it writes certificates
+onto the file system where Zookeeper expectes them and also updates
+certificates once they got changed in Vault allowing rotation without service
+interuption. Agent is configured to use password-less access to Vault.
+
+Establishing trust between CCE and Vault
+========================================
+
+Before any application from the CCE is able to login to vault relying on
+infrastructure based authentication it is required to do some steps on the
+Vault side. Kubernetes authentication plugin is enabled and configured to only
+access requests from specific Kubernetes cluster by providing it's Certificate
+Authority. To allow further multiple different CCE clusters to use vault a
+dedicated auth path is going to be used.
+
+.. code-block:: shell
+
+   $ vault auth enable -path kubernetes_cce1 kubernetes
+   $ vault write auth/kubernetes_cce1/config \
+       kubernetes_host="$K8S_HOST" \
+       kubernetes_ca_cert="$SA_CA_CRT"
+
+Since in our example a dedicated service account with token is being
+periodically rotated using `client JWT as reviewer JWT
+<https://www.vaultproject.io/docs/auth/kubernetes#use-the-vault-client-s-jwt-as-the-reviewer-jwt>`_
+can be used.
+
+Access rules for Vault
+======================
+
+Once Auth plugin is enabled CCE workloads are able to authenticate to Vault,
+but they can do nothing. It is now necessary to establish further level of
+authorization and let particular service account of CCE to get access to
+secrets in vault.
+
+For the scope of the use case we are going to allow zookeeper service account
+from zookeeper namespace to get access to the TLS secrets stored in vault (KV
```

Comment:
KV - key value  ... I think might be good to at least write down the KV = key value